tasks/src/main/java/org/example/aSpyDetected2/ASpyDetected2.java

```diff
@@ -0,0 +1,77 @@
+package org.example.aSpyDetected2;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.StringTokenizer;
+
+/**
+ * https://codeforces.com/problemset/problem/1512/A
+ */
+public class ASpyDetected2 {
+    FastReader sc = new FastReader();
+    int numberOfTests = sc.nextInt();
+
+    void loop() {
+        while (numberOfTests-- > 0) {
+            aSpyDetected2(sc);
+        }
+    }
+
+    public int aSpyDetected2(FastReader sc) {
+        int lengthOfArray = sc.nextInt();
+        int[] arr = new int[lengthOfArray];
+        int unique = 0;
+        int key;
+
+        for (int i = 0; i < lengthOfArray; i++) {
+            int numbersInArray = sc.nextInt();
+            arr[i] = numbersInArray;
+        }
+
+        List<Integer> listOfNumbers = Arrays.stream(arr)
+                .boxed()
+                .sorted()
+                .toList();
+
+        if (listOfNumbers.get(0) == listOfNumbers.get(1)) {
+            key = listOfNumbers.get(listOfNumbers.size() - 1);
```

Comment:
Czy przypadkiem ta lista nie ma tej samej długości co tablica z której powstała?

Reply:
Tak. Chodzi o to, żeby oszczędzić kodu na sortowanie tablicy i szybciej posortować listę. Następnie porównać wartości z przodu i z tyłu i wybrać unikalny numer.

Reply:
To możesz użyć długości tablicy zamiast długości listy. Oszczędzisz czas i pamięć

Reply:
Done.